tensorflow/compiler/xla/literal_util.cc

```diff
@@ -1015,6 +1050,8 @@ static bool AllElementsEqualValue(const Literal& literal, NativeT value) {
       return AllElementsEqualValue<float>(literal, value);
     case F64:
       return AllElementsEqualValue<double>(literal, value);
+    case F16:
+      return AllElementsEqualValue<half>(literal, (half)value);
```

Comment:
strangely this function is checking that they are all equal to an int8.  all the native types support transparent casting between int8 and themselves, but Eigen::half does not.  Shame - it would look prettier.